chameleon/admin.py

```diff
@@ -21,12 +23,35 @@ def pi_eligibility(self):
         return 'Ineligible'
 
 class PIEligibilityAdmin(admin.ModelAdmin):
-    readonly_fields = ['requestor','request_date','reviewer','review_date']
-    fields = ('requestor','request_date','status','review_date','reviewer','review_summary')
+    readonly_fields = ['requestor','request_date','user_metadata', 'reviewer','review_date']
+    fields = ('requestor','request_date','user_metadata','status','review_date','reviewer','review_summary')
     ordering = ['-status','-request_date','-review_date']
 
     list_display = ('requestor','status','request_date')
     list_filter = ('status',)
     search_fields = ['requestor__username']
 
+    def user_metadata(self, obj):
+        keycloak_client = KeycloakClient()
+        keycloak_user = keycloak_client.get_keycloak_user_by_username(obj.requestor.username)
```

Comment:
Keycloak's user attribute structure is a bit wacky. What if we normalize it in this `get_keycloak_user_by_username` function? You could iterate over the attributes, find all the ones that are list of size 1, and flatten them so it's just a key/value pair, as we'd expect. If for some reason the list is size > 1 (multivalued attribute I guess, which we don't utilize at the moment), you could leave it as a list, which I think would be the expected behavior.

If you normalize here, we don't need to have that extractor function, which seems to simplify things a lot.